pkg/apis/v1alpha1/stage_types.go

```diff
@@ -118,6 +118,20 @@ type StageNext struct {
 	Delete bool `json:"delete,omitempty"`
 	// StatusTemplate indicates the template for modifying the status of the resource in the next.
 	StatusTemplate string `json:"statusTemplate,omitempty"`
+	// StatusSubresource indicates the name of the subresource that will be patched.
+	// +default="status"
+	// +kubebuilder:default=status
+	StatusSubresource *string `json:"statusSubresource,omitempty"`
+	// StatusPatchAs indicates the impersonating configuration for client when patching status.
+	// In most cases this will be empty, in which case the default client service account will be used.
+	// When this is not empty, a corresponding rbac change is required to grant `impersonate` privilege.
+	StatusPatchAs *ImpersonationConfig `json:"statusPatchAs,omitempty"`
```

Comment:
The two new fields currently don't support Pod and Node so I feel like I should mention this, or should we add the same support to it?

Reply:
@wzshiming I think we should put this in the field comment and only support in generic stage controller, until there are usecase for Pod and Node to genuinely support it. If it helps, I can add a comment to the field documentation and create an issue for this for future support.

Reply:
updated PTAL @wzshiming